src/test/lib/scheduler/task_queue_test.cpp

```diff
@@ -0,0 +1,31 @@
+#include "base_test.hpp"
+
+#include "scheduler/job_task.hpp"
+#include "scheduler/task_queue.hpp"
+
+using namespace hyrise::expression_functional;  // NOLINT
+
+namespace hyrise {
+
+class TaskQueueTest : public BaseTest {};
+
+TEST_F(TaskQueueTest, StealableJobs) {
+  auto task_queue = TaskQueue{NodeID{0}};
+
+  task_queue.push(std::make_shared<JobTask>([]() { return; }, SchedulePriority::High, true), SchedulePriority::High);
+  EXPECT_TRUE(task_queue.steal());
+
+  task_queue.push(std::make_shared<JobTask>([]() { return; }, SchedulePriority::High, false), SchedulePriority::High);
+  EXPECT_FALSE(task_queue.steal());
+}
+
+TEST_F(TaskQueueTest, EstimateLoad) {
+  auto task_queue = TaskQueue{NodeID{0}};
+
+  task_queue.push(std::make_shared<JobTask>([]() { return; }, SchedulePriority::High), SchedulePriority::High);
+  task_queue.push(std::make_shared<JobTask>([]() { return; }, SchedulePriority::Default), SchedulePriority::Default);
+
+  EXPECT_EQ(task_queue.estimate_load(), size_t{3});
```

Comment:
could you add a comment that roughly explains the expected value (`1 * 2^0 + 1 * 2^1`)?

Reply:
```
  // Tasks of higher priority are weighter higher. Tasks with the default priority have a multiplier of 1, while high
  // priority tasks have a factor of two. Thus we calculate the load as `1 * 2^0 + 1 * 2^1`.
```

Reply:
*weighted, but the overall comment is nice